index.js

```diff
@@ -689,6 +696,47 @@ class Hyperbee extends ReadyResource {
     return (this._checkout <= this.core.length || this._checkout <= 1) ? this.core.snapshot() : this.core.session({ snapshot: false })
   }
 
+  async clearUnlinked (options = {}) {
+    await this.ready()
+
+    const { gte = 0, lt = this.version - 1 } = options
+    const checkout = this.version
+
+    const prev = this.batch({
+      wait: false,
+      checkout: gte
+    })
+
+    const b = this.batch({
```

Comment:
I'd try-finally the code so closing the batches always happens. There are some error paths (though admittedly edge cases).
Examples:
- `getRoot` on a core without header while there's no more disk space available (`core.append` will fail)
- this.core.clear() when out of storage space

Reply:
It will also throw if the core gets closed while the iterator runs, but not closing the batches etc. there might be fine

Reply:
Good call try finally it. Can you pr?